### ISSUE TITLE
Watchfile reloads when Media uploaded to MEDIA_ROOT

### ISSUE THREAD

**Issue by MAkcanca** [cookiecutter/cookiecutter-django#4318]
## What happened?
Whenever I create a file in MEDIA_ROOT (/media) folder, all celery tasks just spit out KeyboardInterrupts along with "change detected". This breaks the Celery chain I built for file processing.
## What should've happened instead?
It shouldn't restart or stop celery tasks when MEDIA_ROOT is touched.
## Additional details
Using latest version cookiecutter-django, pulled fresh.

- Host system configuration:
  - Version of cookiecutter CLI (get it with `cookiecutter --version`): Cookiecutter 2.1.1
  - OS name and version: 
WSL2, Ubuntu, Docker
    On Linux, run

    ```bash
    lsb_release -a 2> /dev/null || cat /etc/redhat-release 2> /dev/null || cat /etc/*-release 2> /dev/null || cat /etc/issue 2> /dev/null
    ```

    On MacOs, run

    ```bash
    sw_vers
    ```

    On Windows, via CMD, run

    ```
    systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
    ```

    ```bash
Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
    ```

  - Python version, run `python3 -V`: Python 3.11.3
  - Docker version (if using Docker), run `docker --version`: Docker version 20.10.23, build 7155243
  - docker-compose version (if using Docker), run `docker-compose --version`: Docker Compose version v2.15.1
  - ...

**Comment by browniebroke** [comment]
Thanks for the report, that sounds like a bug indeed. Happy to review a patch to fix it.

**Comment by foarsitter** [comment]
Can you enable the `watchdog` logger in `settings/base.py` and post the results? 
```
logging.getLogger("watchdog").setLevel(logging.WARNING)
```

It could be related to https://github.com/cookiecutter/cookiecutter-django/issues/4179

**Comment by browniebroke** [comment]
> It could be related to #4179

I don't think it is because the Celery autoreload is implemented using [`watchfiles`](https://watchfiles.helpmanual.io/) (while that issue comes from `watchdog`):

https://github.com/cookiecutter/cookiecutter-django/blob/d3d19264d7236dc704273a555e3741b26d0f848a/%7B%7Bcookiecutter.project_slug%7D%7D/compose/local/django/celery/worker/start#L7

I checked today on one of my projects where the versions ae up to date and was able to reproduce. Looking at [watchfiles' help](https://watchfiles.helpmanual.io/cli/#help), I think we have a couple options to solve this:

1. Add `--ignore-paths media` to ignore anything in the media root
   - Pros: solves this issue and let other source files trigger reload (e.g. templates)
   - Cons: we might trigger the autoreload with other files which shouldn't? Not sure which though...
2. Add `--filter python` to only watch for Python files
   - Pros: solves this issue as well as other potential similar issues with other non-Python files
   - Cons: disables auto-reload for templates, which is _probably_ fine for Celery?

What do you think?

**Comment by MAkcanca** [comment]
I believe Celery workers in Django projects doesn't require real-time updates or changes on Django templates or any non-Python files. Since templates are typically for front-end display and Celery is a task queue for backend processes, it's common that changes in templates wouldn't require a Celery worker restart, so Option 2 sounds more suitable.

**Comment by browniebroke** [comment]
> However, we maybe should document this behavior, because if I update a unrelated view file, my celery tasks will get KeyboardInterrupted and workers will restart. This is also problematic.

I don't think it's that much of a problem. If you change some code, it's quite expected to get the auto-reload to trigger. AFAIU, the OP (reload when media files are uploaded) is about the fact that it could be triggered by a task itself.

> Can we maybe implement a celery tasks directory? Or document this behavior and only track specific files, e.g any file named tasks.py? 

The problem with this approach is that tasks rarely work in isolation, often times they import other code, like your models, utility function or any logic really. But all for adding a paragraph to the docs giving some pointers if further customisation is required. 

I believe that reloading on any changes to Python files is a good compromise, we shouldn't cater for all edge cases but cover the most common ones.
